loader/loader.go

```diff
@@ -377,20 +377,44 @@ func (l *Loader) convertMessage(tspec *ast.TypeSpec) (*desc.DescriptorProto, err
 		if mtype, ok := ftype.(*types.Map); ok {
 			ptype = desc.FieldDescriptorProto_TYPE_MESSAGE
 			plabel = desc.FieldDescriptorProto_LABEL_REPEATED
-			tname, msgNestedType = l.convertMap(tspec.Name.Name, field.Names[0].Name, mtype)
+			tname, msgNestedType = l.convertMap(tspec.Name.Name, fieldName, mtype)
 			msg.NestedType = append(msg.NestedType, msgNestedType)
 		} else {
 			ptype, plabel, tname = l.convertType(ftype)
 		}
 		if ptype == 0 {
 			return nil, fmt.Errorf("unsupported field type: %v", ftype)
 		}
+
+		// Check that the struct field has a tag. We currently
+		// require all struct fields to have a tag; this is used
+		// to assign the position number for a field, ie: `pb:"1"`
+		if field.Tag == nil {
+			return nil, fmt.Errorf("missing required tag on %s", fieldName)
+		}
+		str, err := strconv.Unquote(field.Tag.Value)
```

Comment:
I was skipping the error here on purpose - we've parsed the file, so this can't error.

Reply:
Alright, I'll add a comment so this doesn't confuse me again.